### PR TITLE
fix: warn when linking a package with peerDeps

### LINKED ISSUE #615: `pnpm link` status message
There should be a message after a `pnpm link X` or `pnpm link` command to show what has been linked where.

It should also show a warning that the linked module will not be able to use peer dependencies from the current module (as discussed in https://github.com/pnpm/pnpm/issues/604). It should search the linkee's `package.json` file to check for peer deps.

### PR DESCRIPTION
**Pull request by nachoaldamav**
Closes #615

### CODE REVIEW

**Comment by nachoaldamav** [comment]
Example output from this command:
```bash
nacho@Ignacios-MacBook-Pro foo % pd link ../baz       
 WARN  The package baz, which you have just pnpm linked, has the following peerDependencies specified in its package.json:
        
  - @ialdama/foo@^1.0.0

The linked in dependency will not resolve the peer dependencies from the target node_modules. 
This might cause issues in your project. To resolve this, you may use the "file:" protocol to reference the local dependency.

node_modules:
+ baz 1.0.0 <- ../baz
```

File: pkg-manager/plugin-commands-installation/src/link.ts [hunk 1]
```diff
@@ -122,6 +123,26 @@ export async function handler (
     if (path.relative(linkOpts.dir, cwd) === '') {
       throw new PnpmError('LINK_BAD_PARAMS', 'You must provide a parameter')
     }
+
+    const { manifest: sourceManifest } = await tryReadProjectManifest(linkCwdDir, opts)
+
+    if (sourceManifest?.peerDependencies) {
```
Comment (zkochan) [inline pkg-manager/plugin-commands-installation/src/link.ts]: check that it has at least on dependency in peer dependencies.

File: pkg-manager/plugin-commands-installation/src/link.ts [hunk 2]
```diff
@@ -185,6 +206,28 @@ export async function handler (
...
+        logger.warn({
+          message: `The package ${packageName}, which you have just pnpm linked, has the following peerDependencies specified in its package.json:
+        
+${peerDeps}
+
+The linked in dependency will not resolve the peer dependencies from the target node_modules. 
+This might cause issues in your project. To resolve this, you may use the "file:" protocol to reference the local dependency.`,
+          prefix: opts.dir,
+        })
+      }
```
Comment (zkochan) [inline pkg-manager/plugin-commands-installation/src/link.ts]: this looks like duplicated code. Move it to a function.

**Review by zkochan — COMMENTED** [review]
cover it with a test

**Review by zkochan — APPROVED** [review]

**Pull request merged**